libs/tdigest/src/main/java/org/elasticsearch/tdigest/AVLGroupTree.java

```diff
@@ -29,11 +31,17 @@
 
 import java.util.AbstractCollection;
 import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * A tree of t-digest centroids.
  */
-final class AVLGroupTree extends AbstractCollection<Centroid> implements Releasable {
+final class AVLGroupTree extends AbstractCollection<Centroid> implements Releasable, Accountable {
+    private static final long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(AVLGroupTree.class);
+
+    private final TDigestArrays arrays;
+    private final AtomicBoolean closed = new AtomicBoolean(false);
```

Comment:
Are these thread safe classes other than this? If not I'd prefer we not use `Atomic` stuff. *most* of aggs isn't thread safe and doesn't have to be intentionally.

Reply:
I'm not worried about the overhead, not really. I'm more worried that a reader will see this and say "Oh! this is shared in many threads" and spend some time trying to find that when it isn't really shared.

Reply:
I used AtomicBoolean to replicate the pattern we use along ES (From what I saw, most other classes that check for close() idempotence do that).
But no, those classes aren't thread-safe afaik. So I can change it if it feels better